lib/hyacinth/digital_object/resource.rb

```diff
@@ -3,89 +3,21 @@
 module Hyacinth
   module DigitalObject
     class Resource
-      attr_accessor :import_location, :import_method, :import_checksum,
-                    :location, :checksum, :original_filename, :media_type, :file_size
-
-      attr_reader :import_succeeded
+      LOCATION_FIELD = :location
+      SERIALIZED_FIELDS = [LOCATION_FIELD, :checksum, :original_file_path, :media_type, :file_size].freeze
+      FIELDS = (SERIALIZED_FIELDS + [:is_new]).freeze
+      attr_accessor(*FIELDS)
 
+      # The opts hash should only include symbol keys
       def initialize(opts = {})
         opts.each do |opt_name, opt_value|
-          setter_method = "#{opt_name}=".to_sym
-          self.send(setter_method, opt_value) if self.respond_to?(setter_method)
+          raise ArgumentError, "Invalid option: #{opt_name}" unless FIELDS.include?(opt_name.to_sym)
+          instance_variable_set("@#{opt_name}", opt_value)
         end
-        @import_succeeded = false
```

Comment:
did all this move into an external validation?

Reply:
I guess its the resource_import file further down.

Reply:
Yeah exactly.  A lot of Resource stuff moved to ResourceImport.